qt/scientific_interfaces/Indirect/IndirectSqw.cpp

```diff
@@ -22,189 +22,90 @@ using MantidQt::API::BatchAlgorithmRunner;
-  connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(sqwAlgDone(bool)));
+IndirectSqw::~IndirectSqw() = default;
 
-  connect(m_uiForm.pbRun, SIGNAL(clicked()), this, SLOT(runClicked()));
-  connect(m_uiForm.pbSave, SIGNAL(clicked()), this, SLOT(saveClicked()));
+void IndirectSqw::setup() {}
 
-  connect(this, SIGNAL(updateRunButton(bool, std::string const &, QString const &, QString const &)), this,
-          SLOT(updateRunButton(bool, std::string const &, QString const &, QString const &)));
+/**
+ * Connects the signals in the interface.
+ *
+ */
 
-  m_uiForm.rqwPlot2D->setCanvasColour(QColor(240, 240, 240));
+void IndirectSqw::connectSignals() {
+  connect(m_batchAlgoRunner, SIGNAL(batchComplete(bool)), this, SLOT(sqwAlgDone(bool)));
 
-  // Allows empty workspace selector when initially selected
-  m_uiForm.dsSampleInput->isOptional(true);
+  connect(m_view.get(), SIGNAL(dataReady(QString const &)), this, SLOT(handleDataReady(QString const &)));
+  connect(m_view.get(), SIGNAL(qLowChanged(double)), this, SLOT(qLowChanged(double)));
+  connect(m_view.get(), SIGNAL(qWidthChanged(double)), this, SLOT(qWidthChanged(double)));
+  connect(m_view.get(), SIGNAL(qHighChanged(double)), this, SLOT(qHighChanged(double)));
+  connect(m_view.get(), SIGNAL(eLowChanged(double)), this, SLOT(eLowChanged(double)));
+  connect(m_view.get(), SIGNAL(eWidthChanged(double)), this, SLOT(eWidthChanged(double)));
+  connect(m_view.get(), SIGNAL(eHighChanged(double)), this, SLOT(eHighChanged(double)));
+  connect(m_view.get(), SIGNAL(rebinEChanged(int)), this, SLOT(rebinEChanged(int)));
 
-  // Disables searching for run files in the data archive
-  m_uiForm.dsSampleInput->isForRunFiles(false);
-}
+  connect(m_view.get(), SIGNAL(runClicked()), this, SLOT(runClicked()));
+  connect(m_view.get(), SIGNAL(saveClicked()), this, SLOT(saveClicked()));
 
-IndirectSqw::~IndirectSqw() = default;
+  connect(m_view.get(), SIGNAL(showMessageBox(const QString &)), this, SIGNAL(showMessageBox(const QString &)));
 
-void IndirectSqw::setup() {}
+  connect(this, SIGNAL(updateRunButton(bool, std::string const &, QString const &, QString const &)), m_view.get(),
+          SLOT(updateRunButton(bool, std::string const &, QString const &, QString const &)));
+}
 
 /**
  * Handles the event of data being loaded. Validates the loaded data.
  *
  */
 void IndirectSqw::handleDataReady(QString const &dataName) {
-  UserInputValidator uiv;
-  validateDataIsOfType(uiv, m_uiForm.dsSampleInput, "Sample", DataType::Red);
-
-  auto const errorMessage = uiv.generateErrorMessage();
-  if (!errorMessage.isEmpty()) {
-    showMessageBox(errorMessage);
-  } else {
-    plotRqwContour(dataName.toStdString());
-    setDefaultQAndEnergy();
+  if (m_view->validate()) {
+    m_model->setInputWorkspace(dataName.toStdString());
+    m_model->setEFixed(getInstrumentDetail("Efixed").toStdString());
+    plotRqwContour();
+    m_view->setDefaultQAndEnergy();
```

Comment:
does this mean we no longer produce the warning if the data is of the wrong type?